src/source-af-packet.c

```diff
@@ -1977,16 +1977,15 @@ static int AFPSetupRing(AFPThreadVars *ptv, char *devname)
 /** \brief test if we can use FANOUT. Older kernels like those in
  *         CentOS6 have HAVE_PACKET_FANOUT defined but fail to work
  */
-int AFPIsFanoutSupported(int cluster_id)
+int AFPIsFanoutSupported(uint16_t cluster_id)
 {
 #ifdef HAVE_PACKET_FANOUT
     int fd = socket(AF_PACKET, SOCK_RAW, htons(ETH_P_ALL));
     if (fd < 0)
         return 0;
 
     uint32_t mode = PACKET_FANOUT_HASH | PACKET_FANOUT_FLAG_DEFRAG;
-    uint16_t id = 1;
-    uint32_t option = (mode << 16) | (id & 0xffff);
+    uint32_t option = (mode << 16) | (cluster_id & 0xffff);
```

Comment:
What is the effect of `<u16> & 0xffff`? Would it ever be something else than just `<u16>`?

Reply:
Ah, no.. With the type of the function changed that could not be removed.